pkg/apis/kudo/v1beta1/instance_types_helpers.go

```diff
@@ -196,6 +205,35 @@ func (i *Instance) IsTopLevelInstance() bool {
 	return !i.IsChildInstance()
 }
 
+const (
+	planInProgressReason   = "PlanInProgress"
+	planInFatalErrorReason = "PlanInFatalError"
+	resourceNotReadyReason = "ResourceNotReady"
+	resourcesReadyReason   = "ResourcesReady"
+	readyConditionType     = "Ready"
+)
+
+func (i *Instance) SetReadinessTrue(msg string) {
+	condition := metav1.Condition{Type: readyConditionType, Status: metav1.ConditionTrue, Message: msg, Reason: resourcesReadyReason}
+	meta.SetStatusCondition(&i.Status.Conditions, condition)
+}
+
+func (i *Instance) SetNotReady(msg string) {
+	condition := metav1.Condition{Type: readyConditionType, Status: metav1.ConditionFalse, Message: msg, Reason: resourceNotReadyReason}
+	meta.SetStatusCondition(&i.Status.Conditions, condition)
+}
+
+func (i *Instance) SetReadinessFatalError(msg string) {
+	condition := metav1.Condition{Type: readyConditionType, Status: metav1.ConditionFalse, Message: msg, Reason: planInFatalErrorReason}
```

Comment:
Wait, we had a **really** long slack discussion were you explained why a `FATAL_ERROR` should be treated as `Unknown` but now it is `False` anyway?

Reply:
good catch :D this is of course a bug :D